onnxruntime/contrib_ops/cpu/triu.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+#include "core/platform/threadpool.h"
+#include "core/util/math_cpuonly.h"
+#include "Eigen/src/Core/Map.h"
+#include "Eigen/LU"
+#include "triu.h"
+#include <functional>
+
+namespace onnxruntime {
+namespace contrib {
+
+ONNX_OPERATOR_TYPED_KERNEL_EX(
+    Triu,
+    kMSDomain,
+    1,
+    float,
+    kCpuExecutionProvider,
+    KernelDefBuilder()
+        .TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
+    Triu);
+
+ONNX_OPERATOR_TYPED_KERNEL_EX(
+    Triu,
+    kMSDomain,
+    1,
+    double,
+    kCpuExecutionProvider,
+    KernelDefBuilder()
+        .TypeConstraint("T", DataTypeImpl::GetTensorType<double>()),
+    Triu);
+
+template <typename T>
+struct Triu::ComputeImpl {
+
+  void get_triu(const T* X_data, T* Y_data, int64_t matrix_h, int64_t matrix_w, int64_t k_val) const {
```

Comment:
The spacing looks off here: run clang-format to cleanup?

Reply:
How do you run this for ORT repo?

Reply:
I use clang-format from MSVC and do "clang-format triu.cc >out", diff the files to make sure it did what I expected, then copy the temp file back over the original file. The ORT rules for clang-format are in the root of the repo and are picked up automatically by clang-format.